ledger/src/blocktree_processor.rs

```diff
@@ -412,77 +412,174 @@ pub fn verify_ticks(
+        if load_result.is_err() {
+            timing.fetch_fail_elapsed += load_elapsed.as_us();
+        } else {
+            timing.fetch_elapsed += load_elapsed.as_us();
+        }
+        load_result
+    }?;
+
+    let num_entries = entries.len();
+    let num_txs = entries.iter().map(|e| e.transactions.len()).sum::<usize>();
+    trace!(
+        "Fetched entries for slot {}, num_entries: {}, num_shreds: {}, num_txs: {}, slot_full: {}",
+        slot,
+        num_entries,
+        num_shreds,
+        num_txs,
+        slot_full,
+    );
 
-    if entry_state.status() == EntryVerificationStatus::Failure {
-        warn!("Ledger proof of history failed at slot: {}", bank.slot());
-        return Err(BlockError::InvalidEntryHash.into());
+    if !skip_verification {
+        let tick_hash_count = &mut progress.tick_hash_count;
+        verify_ticks(bank, &entries, slot_full, tick_hash_count).map_err(|err| {
+            warn!(
+                "{:#?}, slot: {}, entry len: {}, tick_height: {}, last entry: {}, last_blockhash: {}, shred_index: {}, slot_full: {}",
+                err,
+                slot,
+                num_entries,
+                bank.tick_height(),
+                progress.last_entry,
+                bank.last_blockhash(),
+                num_shreds,
+                slot_full,
+            );
+            err
+        })?;
     }
 
-    let mut processing_elapsed = Measure::start("processing_elapsed");
+    let mut verify_total = Measure::start("verify_and_process_entries");
+    let verifier = if !skip_verification {
+        datapoint_debug!("verify-batch-size", ("size", num_entries as i64, i64));
+        let entry_state = entries.start_verify(&progress.last_entry, recyclers.clone());
+        if entry_state.status() == EntryVerificationStatus::Failure {
+            warn!("Ledger proof of history failed at slot: {}", slot);
+            return Err(BlockError::InvalidEntryHash.into());
+        }
+        Some(entry_state)
+    } else {
+        None
+    };
+
+    let mut replay_elapsed = Measure::start("replay_elapsed");
     let process_result = process_entries_with_callback(
         bank,
-        entries,
+        &entries,
         true,
         entry_callback,
         transaction_status_sender,
     )
     .map_err(BlocktreeProcessorError::from);
-    processing_elapsed.stop();
+    replay_elapsed.stop();
+    timing.replay_elapsed += replay_elapsed.as_us();
 
-    if !entry_state.finish_verify(entries) {
-        warn!("Ledger proof of history failed at slot: {}", bank.slot());
-        return Err(BlockError::InvalidEntryHash.into());
+    if let Some(mut verifier) = verifier {
+        if !verifier.finish_verify(&entries) {
+            warn!("Ledger proof of history failed at slot: {}", bank.slot());
+            return Err(BlockError::InvalidEntryHash.into());
+        }
+        verify_total.stop();
+        timing.verify_elapsed += verify_total.as_us() - replay_elapsed.as_us();
```

Comment:
Maybe we should rename `verify_elapsed` to `additonal_verify_elapsed`, as it communicates a relative time frame to `replay_elapsed`, not the full verification time

Reply:
Hmm it does represent the full verification time if we verify with CPU's. It's only when we use GPU's that we are parallelizing the verification. How about I add the parallelized gpu processing time to get a true verify time here?